csharp/src/Google.Protobuf.Test/GeneratedMessageTest.cs

```diff
@@ -691,6 +691,54 @@ public void OneofSerialization_DefaultValue()
             });
         }
 
+        [Test]
+        public void MapSerializationDeterministicTrue_ThenBytesIdentical()
```

Comment:
Will add more tests for different types of keys, once the general implementation approach is approved.